rotkehlchen/exchanges/bybit.py

```diff
@@ -0,0 +1,514 @@
+                    asset=coin,
+                    amount=deserialize_asset_amount(movement['amount']),
+                    fee_asset=coin,
+                    fee=Fee(ZERO),
+                    link=movement['txID'],
+                ))
+            except DeserializationError as e:
+                log.error(f'{e} when reading movement for bybit trade {movement}')
+                continue
+
+        return movements
+
+    def _query_withdrawals(self, options: dict[str, Any]) -> list[AssetMovement]:
+        """
+        Process withdrawals from bybit. If any asset is unknown or we fail to deserialize an entry
+        the error is logged and we skip the entry.
+        """
+        log.debug(f'querying online withdrawals for {options}')
+        raw_data = self._paginated_api_query(
+            endpoint='asset/withdraw/query-record',
+            options=options,
+        )
+        movements = []
+        for movement in raw_data:
+            try:
+                coin = asset_from_bybit(movement['coin'])
+            except UnknownAsset:
+                log.error(f'Could not read assets from withdrawal {movement["coin"]}')
+                continue
+
+            try:
+                movements.append(AssetMovement(
+                    timestamp=ts_ms_to_sec(TimestampMS(int(movement['updateTime']))),
+                    location=Location.BYBIT,
+                    category=AssetMovementCategory.WITHDRAWAL,
+                    address=None,
+                    transaction_id=movement['txID'],
+                    asset=coin,
+                    amount=deserialize_asset_amount(movement['amount']),
+                    fee_asset=coin,
+                    fee=Fee(deserialize_asset_amount(movement['withdrawFee'])),
+                    link=movement['withdrawId'],
+                ))
+            except DeserializationError as e:
+                log.error(f'{e} when reading movement for bybit {movement}')
+                continue
+
+        return movements
+
+    def query_online_deposits_withdrawals(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> list[AssetMovement]:
+        new_movements = []
+        upper_ts, lower_ts = end_ts, Timestamp(end_ts - DAY_IN_SECONDS * 30)
+        two_years_ago = Timestamp(ts_now() - DAY_IN_SECONDS * 365 * 2)
+        if start_ts < two_years_ago:
+            start_ts = two_years_ago
+
+        while True:
+            options = {
+                'endTime': str(ts_sec_to_ms(upper_ts)),
+                'limit': PAGINATION_LIMIT,
+                'startTime': str(ts_sec_to_ms(lower_ts)),
+            }
+
+            new_movements.extend(self._query_deposits(options.copy()))
+            new_movements.extend(self._query_withdrawals(options.copy()))
+
+            upper_ts = Timestamp(upper_ts - DAY_IN_SECONDS * 30)
+            lower_ts = Timestamp(lower_ts - DAY_IN_SECONDS * 30)
+
+            if upper_ts <= start_ts:
+                break
```

Comment:
too many newlines ... also why copy the options? Why 30 days? Explain what all these stuff are

```suggestion
            new_movements.extend(self._query_deposits(options.copy()))
            new_movements.extend(self._query_withdrawals(options.copy()))
            upper_ts = Timestamp(upper_ts - DAY_IN_SECONDS * 30)
            lower_ts = Timestamp(lower_ts - DAY_IN_SECONDS * 30)
            if upper_ts <= start_ts:
                break
```